lib/TH/generic/THTensorLapack.c

```diff
@@ -418,6 +418,9 @@ void THTensor_(gesvd2)(THTensor *ru_, THTensor *rs_, THTensor *rv_, THTensor *ra
                                THTensor_(free)(work);),
                            "gesvd", info);
 
+  if (*jobu == 'S')
+    THTensor_(narrow)(rv__,NULL,1,0,k);
```

Comment:
instead of doing this, wouldn't you initialize rv_ / rv__ to only be k elements in the first place, instead of allocating a bigger tensor and then narrowing (which doesn't actually save memory because this tensor is just a narrow into a larger storage).